proposals/NNNN-Addition-of-a-way-to-notify-HMI-of-the-required-support-for-apps.md

```diff
@@ -0,0 +1,103 @@
+
+#### HMI_API
+The HMI_API `requestAdditionalSupport` has two parameters `appName` and `requiredAdditionalSupport`. Also, the HMI_API `pingApps` has a parameters `appName`. The description of these parameter is shown as below.
+
+```xml
++   <function name="RequestAdditionalSupport" messagetype="notification">
++       <description>Notification from SDL to HMI to notify the additional supports that SDL app needs.</description>
++       <param name="appName" type="String" maxlength="100" mandatory="true">
++           <description>The name of SDL app that needs the additional supports.</description>
++       </param>
++       <param name="requiredAdditionalSupport" type="String" minsize="0" maxsize="100" maxlength="100" array="true" mandatory="true">
++           <description>An array of the additional suuport names which SDL app needs. Ex, audioSupport, highBandwidth.</description>
++       </param>
++   </function>
+
++   <function name="PingApps" messagetype="notification">
++       <description>Notification from HMI to SDL to notify that SDL app is available the additional supports.</description>
++       <param name="appName" type="String" maxlength="100" mandatory="true">
++           <description>The name of SDL app that uses the additional supports.</description>
++       </param>
++   </function>
+```
+
+#### Modification of Router Service
+To realize this proposal, it is necessary to modify the Router service as follows.
+ - Receive the `Ping Apps` message from SDL Core at Protocol Spec.
+ - Search the target SDL app that have the name of SDL app stored in the parameter `appName` of `Ping Apps` message.
+ - Send the `SDL_CONNECTED` message to the target SDL app.
+
+#### Sequence diagram of the way to notify HMI of the required support for apps
+Figure 1 shows the sequence the way to notify HMI of the required support for apps.
+
+<b>Figure 1.</b> Sequence diagram of the way to notify HMI of the required support for apps
+
+![Figure_1_sequence_of_the_way_to_notify_hmi_of_the_support.png](../assets/proposals/NNNN-Addition-of-a-way-to-notify-HMI-of-the-required-support-for-apps/Figure_1_sequence_of_the_way_to_notify_hmi_of_the_support.png)
+
+The sequence describes as follows.
+1. SDL app confirms whether to be enough the support that it needs.
+2. SDL app sends HU (SDL Core) the `Request Additional Support` message at Protocol spec if the required support is not enough.
+3. SDL Core sends HMI the RPC `RequestAdditionalSupport`.
+4. HMI displays to the user a message that support is enough.
+5. User meets support according to displayed message.
+6. If HMI detected to be met the support, it sends SDL Core the RPC `PingApps`.
+7. SDL Core sends SDL app (Router service) the `Ping Apps` message at Protocol Spec.
+8. SDL app (Router service) searches the target SDL app that have the name of SDL app stored in the parameter `appName` of `Ping Apps` message.
+9. SDL app (Router service) sends target SDL app the `SDL_CONNECTED` message.
```

Comment:
Can you please clarify the meaning of "SDL app (Router service)". These are not the same thing.

Reply:
"SDL app (Router service)" means the Router service in app library.
Therefore, we will modify like below.
7. SDL Core sends Router service in the app library the Protocol Spec message `Ping Apps`.
8. Router service searches the target SDL app that have the name of SDL app stored in the parameter `appName` of `Ping Apps` message.
9. Router service sends target SDL app the `SDL_CONNECTED` message.